tapgarden/custodian_test.go

```diff
@@ -584,12 +584,26 @@ func TestBookAssetSyncer(t *testing.T) {
 	// If we add the asset to the asset syncer, address creation should
 	// succeed.
 	h.syncer.AddAsset(*newAsset)
-	addrVersion = test.RandFlip(address.V0, address.V1)
-	newAddr, err := h.addrBook.NewAddress(
-		ctx, addrVersion, newAsset.ID(), 1, nil, proofCourierAddr,
-	)
+
+	// Fetch the asset from the syncer. This should trigger the
+	// background goroutine to add the asset to the address book.
+	_, err = h.syncer.FetchAsset(newAsset.ID())
 	require.NoError(t, err)
-	require.NotNil(t, newAddr)
+
+	// Eventually, the asset should be registered and we should be able to
+	// create a new address for it.
+	var newAddr *address.AddrWithKeyInfo
+	addrVersion = test.RandFlip(address.V0, address.V1)
+
+	require.Eventually(t, func() bool {
+		newAddr, err = h.addrBook.NewAddress(
+			ctx, addrVersion, newAsset.ID(), 1, nil,
+			proofCourierAddr,
+		)
+		require.NoError(t, err)
```

Comment:
What's the actual failure case here if the sync didn't happen yet? I would expect there to be an error. Or will the address actually be `nil`? If the first, then we probably need to do:
```
if err != nil {
    return false
}

return newAddr != nil
```

Reply:
Good point. For some reason I thought `t` would work differently within the callback, but of course, it does not.